src/app/Services/validators.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable, inject } from '@angular/core';
+import { UserService } from './user.service';
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ValidatorsService {
+  userService: UserService = inject(UserService);
```

Comment:
Bhai its better to use construction for DI for better readability and understanding.
The inject() method allows us to manually retrieve dependencies from the DI container. This approach might be useful in situations where we need to conditionally inject dependencies or dynamically resolve dependencies at runtime.

Reply:
updated thanks bhai.

Reply:
thanks bhai, updated